src/texteff.cpp

```diff
@@ -121,7 +121,7 @@ void DrawTextEffects(DrawPixelInfo *dpi)
 
 	for (TextEffect &te : _text_effects) {
 		if (te.string_id == INVALID_STRING_ID) continue;
-		if (te.mode == TE_RISING || (_settings_client.gui.loading_indicators && !IsTransparencySet(TO_LOADING))) {
+		if ((te.mode == TE_RISING && !IsTransparencySet(TO_TEXT)) || (_settings_client.gui.loading_indicators && !IsTransparencySet(TO_TEXT))) {
```

Comment:
Minor detail, but this conditional is somewhat confusing because of the short-circuit evaluation and double check of transparency.
```suggestion
		if (IsTransparencySet(TO_TEXT)) continue;
		if (te.mode == TE_RISING || _settings_client.gui.loading_indicator) {
```

Reply:
Much nicer :)